Move setting color and depth stencil targets to binding.

maoliver-amd opened the pull request:
This builds upon #95 with additional requested changes.

gboisse commented:
I'm getting the following error when trying to build in `Debug`:
```
Error C2679 binary '[': no operator found which takes a right-hand operand of type 'GfxInternal::RenderTarget' (or there is no acceptable conversion) C:\Users\Admin\Documents\gfx\gfx.h 6042
```

Also, the following error messages are spamming the console output when running the `02-pbr` example:
```
[ments\gfx\gfx.h:3345] Error: Cannot draw to an invalid texture object (0x2: Invalid operation)
```

maoliver-amd commented:
> I'm getting the following error when trying to build in `Debug`:
> 
> ```
> Error C2679 binary '[': no operator found which takes a right-hand operand of type 'GfxInternal::RenderTarget' (or there is no acceptable conversion) C:\Users\Admin\Documents\gfx\gfx.h 6042
> ```

Typo in an assert statement, fixed now

> Also, the following error messages are spamming the console output when running the `02-pbr` example:
> 
> ```
> [ments\gfx\gfx.h:3345] Error: Cannot draw to an invalid texture object (0x2: Invalid operation)
> ```

Hmm, I had just assumed Sylvain had properly checked this code when he wrote it up. Fixed now by just removing it as its not needed.
